Sources/XCHammer/XcodeTarget.swift

```diff
@@ -312,17 +312,25 @@ public class XcodeTarget: Hashable, Equatable {
             return []
         }
 
-        return unfilteredDependencies.flatMap { (xcodeTarget: XcodeTarget) -> [XcodeTarget] in
+        var transitiveTargets: [XcodeTarget] = []
+        var queue = unfilteredDependencies
+        while let xcodeTarget = queue.first {
```

Comment:
It looks like github lost my review comment:

With the current impl, we = visit an `XcodeTarget` more than once - for some dep graphs, this has terrible complexity and I can imagine would be slow. Consider this graph:

```
        a
      /   \
    b      c
  /          
c
```
Since order doesn't matter, we _should_ be able to visit a target once.

Additionally, it might be beneficial to memoize this function, as it's called a handful of times with the same arguments.

Given this only costs `.7` seconds, it seems pretty good for now 👍 🎉

Reply:
> it might be beneficial to memoize this function

good call

Reply:
#82 addresses this one, thanks for following up @kastiglione 😄 , this seems to have really sped up the sample and I'm excited to see what it does for our gen times as well.